WIP: Add Freesurfer 7

Shotgunosine opened the pull request:
I added Freesurfer 7 and some additional jobs on circleci to test and deploy it. Circlci config will likely need to be fixed before this is complete.

Shotgunosine committed:
make freesurfer 7 docker file

Commit:
bump to FS 7.1.1

Shotgunosine committed:
Merge pull request #1 from AndysWorth/fs_7

ENH: bump to FS 7.1.1

Shotgunosine committed:
add 7.1.1

Shotgunosine committed:
fix circleci config

Shotgunosine committed:
have fs_7 tests use fs_7 image

Shotgunosine committed:
fix don't use python2 prefix to group commands if we're running freesurfer 7

Shotgunosine commented:
@PeerHerholz and @andysworth this resolves #57 with the potential exception of the replacement recon-all script. I'd be happy to get your feedback on it.

PeerHerholz commented:
Ahoi hoi @Shotgunosine,

thx for the fantastic work and effort. The same to you @andysworth.
I checked the code and tests, LGTM.  The plan re the `recon-all` script would be include it once we have it, eh?

Shotgunosine commented:
Yeah, That's what I'm thinking.

PeerHerholz commented:
Awesome! @andysworth, do you want to have a look as well? After that or otherwise, we could go ahead and merge!?

AndysWorth commented:
Sorry for the slow response.  This looks good to me.  I just noticed that I got a response on the Freesurfer mailing list from Doug Greve:

> Hi Andy, this is a bug introduced in 7.1.1, some kind of race condition.
> You can fix it by commenting out this line in recon-all
>   if($OMP_NUM_THREADS > 1 && ! $LHonly && ! $RHonly) set cmd = ($cmd --parallel) # just hemis
> doug

Thanks for your strong work @Shotgunosine!

PeerHerholz commented:
Cool, thanks @AndysWorth! @Shotgunosine should we go ahead and merge? WDYT re the `recon-all` script adaptation above?

Shotgunosine commented:
We should add a sed command to the docker command, probably the best place
would be in neurodocker. @AndysWorth, do you think you could open an issue
there and submit a PR?

On Fri, Nov 13, 2020 at 8:42 AM Peer Herholz <notifications@github.com>
wrote:

> Cool, thanks @AndysWorth <https://github.com/AndysWorth>! @Shotgunosine
> <https://github.com/Shotgunosine> should we go ahead and merge? WDYT re
> the recon-all script adaptation above?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BIDS-Apps/freesurfer/pull/67#issuecomment-726771234>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABILLGBJ27Y7SRH3TGMV4MTSPUZU7ANCNFSM4TI2FGUA>
> .
>

AndysWorth commented:
Will do!

PeerHerholz commented:
Hi folks,

just checking: we can merge this, right? @Shotgunosine @AndysWorth

AndysWorth commented:
Sorry, I was looking forward to making that contribution but it seems that I'm slowing this down.  It looks like I won't have time until I'm off work for the holidays.  @Shotgunosine if you want to go ahead with your suggestion in the meanwhile, that would be great.  I know it's a small change but I don't feel confident to just do it without taking the time to make sure I fully understand what's going on.

Shotgunosine commented:
I might not be able to get to it until around the same time period. Maybe @llevitis would be able to take a look

AndysWorth commented:
I've installed the fix here at Flywheel -- once I have had a chance to run it a few times I'll submit it to neurodocker.

Shotgunosine commented:
@AndysWorth have you had a chance to send the fix over to neurodocker?

AndysWorth commented:
@Shotgunosine Sorry, no 😞 .  There are two places that need patching in recon-all and the fixes have worked well at Flywheel so it's about time!  I'll work on making the time.  Thanks for the reminder.

Shotgunosine commented:
@AndysWorth, sounds good. It'd be great if you're able to make the changes this week, I've got a bunch of sessions I need to run and I'd love to be able to use this.

AndysWorth commented:
Okay @Shotgunosine, I will!

AndysWorth commented:
Hey @Shotgunosine , I (finally) managed to patch Freesurfer 7.7.1 in neurodocker.  See [neurodocker PR#384](https://github.com/ReproNim/neurodocker/pull/384).  I did this exact same change to the official Freesurfer docker image version of 7.1.1 and it has been run successfully many times on various Flywheel instances.  I _have not_ tested this new neurodocker version, which is for a slightly different version of centos (6 vs. 7).  Is this PR sufficient for you to give it a try?  All it should do is prevent crashing.

AndysWorth commented:
@Shotgunosine I've created a repository [recon-all-v7.1.1-parallel-patch](https://github.com/AndysWorth/recon-all-v7.1.1-parallel-patch).  Right now it just has the patch file that can be installed with
`patch /usr/local/freesurfer/bin/recon-all parallel.patch`
Is this enough?  If not, what else should I do?

Shotgunosine commented:
@AndysWorth, can you update the neurodocker PR to pull that repo and apply the patch?

AndysWorth commented:
@Shotgunosine I'm not familiar enough with neurodocker.  Where would I do that?

AndysWorth commented:
@Shotgunosine It sounds like they don't want the patch in neurodocker so the following should go somewhere around [here](https://github.com/Shotgunosine/freesurfer/blob/fs_7/generate_freesurfer_images.sh#L96)?

```
curl -LJO https://github.com/AndysWorth/recon-all-v7.1.1-parallel-patch/tarball/master
tar zxf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe.tar.gz
cd AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
patch $FREESURFER_HOME/bin/recon-all parallel.patch 
patch $FREESURFER_HOME/bin/quantifyThalamicNuclei.sh quantifyThalamicNuclei.patch
cd ..
rm -rf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
```

Shotgunosine commented:
I’ll see if I can use the script you wrote to get a command into
neurodocker that’ll make them happy.

On Thu, Apr 1, 2021 at 4:57 PM Andrew J. Worth ***@***.***>
wrote:

> @Shotgunosine <https://github.com/Shotgunosine> It sounds like they don't
> want the patch in neurodocker so the following should go somewhere around
> here
> <https://github.com/Shotgunosine/freesurfer/blob/fs_7/generate_freesurfer_images.sh#L96>
> ?
>
> curl -LJO https://github.com/AndysWorth/recon-all-v7.1.1-parallel-patch/tarball/master
> tar zxf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe.tar.gz
> cd AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
> patch $FREESURFER_HOME/bin/recon-all parallel.patch
> patch $FREESURFER_HOME/bin/quantifyThalamicNuclei.sh quantifyThalamicNuclei.patch
> cd ..
> rm -rf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BIDS-Apps/freesurfer/pull/67#issuecomment-812168436>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABILLGC46EKVCXGLUICFAPTTGTM4FANCNFSM4TI2FGUA>
> .
>

Shotgunosine commented:
With the new clarifications can you take care of it @Andysworth?

On Thu, Apr 1, 2021 at 5:15 PM Dylan Nielson ***@***.***> wrote:

> I’ll see if I can use the script you wrote to get a command into
> neurodocker that’ll make them happy.
>
> On Thu, Apr 1, 2021 at 4:57 PM Andrew J. Worth ***@***.***>
> wrote:
>
>> @Shotgunosine <https://github.com/Shotgunosine> It sounds like they
>> don't want the patch in neurodocker so the following should go somewhere
>> around here
>> <https://github.com/Shotgunosine/freesurfer/blob/fs_7/generate_freesurfer_images.sh#L96>
>> ?
>>
>> curl -LJO https://github.com/AndysWorth/recon-all-v7.1.1-parallel-patch/tarball/master
>> tar zxf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe.tar.gz
>> cd AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
>> patch $FREESURFER_HOME/bin/recon-all parallel.patch
>> patch $FREESURFER_HOME/bin/quantifyThalamicNuclei.sh quantifyThalamicNuclei.patch
>> cd ..
>> rm -rf AndysWorth-recon-all-v7.1.1-parallel-patch-e5ff2fe
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/BIDS-Apps/freesurfer/pull/67#issuecomment-812168436>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABILLGC46EKVCXGLUICFAPTTGTM4FANCNFSM4TI2FGUA>
>> .
>>
>

bpinsard commented:
Hi, are there any blockers to merging that very old PR? 
It would be great to have latest versions integrated into repronim/containers. 
Anything I can do to help?
Should we restart a PR from https://github.com/e-dads/freesurfer ?
Thanks!

Shotgunosine commented:
I haven't looked at this in a while, but I think this convo with Satra is where things were left off:

https://github.com/ReproNim/neurodocker/pull/384#issuecomment-812187770

@bpinsard, are you comfortable with the structure of everything to just make that change? It'd take me a while to reestablish context here.

Shotgunosine commented:
Actually, looks like we could just jump to Freesurfer 7.4.1. Would that be alright?

bpinsard commented:
I am not sure I have played enough with neurodocker to fully grasp the issue to be fixed. 
If I understand, that would mean: 
- inserting the patch in your branch as @satra suggested into that file: https://github.com/Shotgunosine/freesurfer/blob/fs_7/generate_freesurfer_images.sh 
- regenerating Dockerfile and Apptainer recipes.
The latest version would be great! I don't think there is really an interest in redeploying each older version.

Remi-Gau commented:
I can have a look to try to fix some of the merge conflicts on the circle-CI config. I have tried to "standardize" across bids apps to make the maintenance easier.

Shotgunosine commented:
I'm working on 7.4.1 now, to do that I've got to update the neurodocker version, which breaks some of the syntax. I'm working on it though.

bpinsard commented:
I had the [generate_freesurfer_images.sh](https://github.com/Shotgunosine/freesurfer/blob/fs_7/generate_freesurfer_images.sh) running with the following changes:
https://github.com/courtois-neuromod/freesurfer/tree/pr/67 
If I understand correctly the patch are not needed with that version, right?

Shotgunosine committed:
bump neurodocker to master and freesurfer to 7.4.1

Shotgunosine committed:
ubunto to jammy

Shotgunosine commented:
@Remi-Gau I took a first pass at resolving merge conflicts on the circleci config, but I've been out of the circleci game for a minute. Do you think you can fix the config now, or have I just made a mess of things?

Remi-Gau commented:
had a quick look
it seems to make sense but may need to go for a deeper dive: those diffs are hard to read

Remi-Gau commented:
also seems that some file changes are showing in the PR when they are already in the default branch, no?

Shotgunosine commented:
I tried to rebase the PR thinking that that would be more straightfoward, but may have just made a mess. Happy to do a force rollback if you think that'd be easier.

Remi-Gau commented:
yeah if you can try to get a cleaner diff it'd be great

let me know if I can help

Shotgunosine committed:
merge master

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Shotgunosine commented:
Alright, should be cleaner now.

Remi-Gau commented:
yup indeed

Remi-Gau commented on .circleci/config.yml:
```diff
@@ -171,6 +171,99 @@ jobs:
+        name: load image
+        command: docker load -i ~/docker/image.tar
+    - run:
+        name: print version
+        command: |
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v $PWD/ds114_test1:/bids_dataset bids/${CIRCLE_PROJECT_REPONAME}_fs7 --version
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test1:/bids_dataset bids/${CIRCLE_PROJECT_REPONAME}_fs7 -h
+    - run:
+        name: test 1
+        command: |
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test1:/bids_dataset -v ${HOME}/outputs1:/outputs bids/${CIRCLE_PROJECT_REPONAME}_fs7 /bids_dataset /outputs participant --participant_label 01 --license_file=/license.txt --stages autorecon1 && cat ${HOME}/outputs1/sub-01/scripts/recon-all.done
+        no_output_timeout: 6h
+    - run:
+        name: test 2
+        command: |
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test2:/bids_dataset -v ${HOME}/outputs2:/outputs bids/${CIRCLE_PROJECT_REPONAME}_fs7 /bids_dataset /outputs participant --participant_label 01 --steps cross-sectional --session_label test --license_file=/license.txt --stages autorecon1 && cat ${HOME}/outputs2/sub-01_ ...
+        no_output_timeout: 6h
+    - run:
+        name: group2 test 1
+        command: |
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test1:/bids_dataset -v ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0.0:/outputs bids/${CIRCLE_PROJECT_REPONAME}_fs7 /bids_dataset /outputs group2 --license_file=/license.txt  && mkdir -p ${HOME}/outputs1/ && sudo mv ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0 ...
+        no_output_timeout: 6h
+    - run:
+        name: group2 test 2
+        command: |
+          docker run -ti --rm --read-only -v ~/license/license.txt:/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test2:/bids_dataset -v ${HOME}/data/ds114_test2_freesurfer_precomp_v6.0.0:/outputs bids/${CIRCLE_PROJECT_REPONAME}_fs7 /bids_dataset /outputs group2 --license_file=/license.txt  && mkdir -p ${HOME}/outputs2/ && sudo mv ${HOME}/data/ds114_test2_freesurfer_precomp_v6.0 ...
```
do you mind if we "multiline" those commands?

I find docker calls super hard to parse in general so I try to stick to one argument, parameter per line.

See here for example: https://github.com/bids-apps/freesurfer/blob/6b76873f65dd5c027c54e3440bd1ae9423e462f6/.circleci/config.yml#L102-L115

Shotgunosine replied:
sure, I'll clean those up.

Remi-Gau replied:
thanks

Remi-Gau commented:
also just thinking: if we agree that we want both freesurfer 6 and 7 to be available it should probably be mentioned in the readme how to pull each from docker hub.
Does not have to be part of this PR: can be done once a release has been done and a tagged image has been pushed

Shotgunosine committed:
reformat docker calls in test_fs7

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Shotgunosine commented:
I think we will want both freesurfer 6 and 7 available. HCP still uses freesurfer 6.

Remi-Gau commented:
having a look locally at the whole config.

it looks very much like this is doing exactly the same thing for both images, no?

like literally running the same commands with both containers

sounds like this should probably be refactored, no?

Shotgunosine commented:
Definitely, do you think you could refactor it? It should be straightforward, I just don't know circle's syntax very well.

Remi-Gau commented:
> Definitely, do you think you could refactor it? It should be straightforward, I just don't know circle's syntax very well.

yup I will give it a go

Remi-Gau committed:
update recipes

Remi-Gau committed:
refactor

Remi-Gau committed:
refactor

Remi-Gau committed:
refactor

Remi-Gau commented:
ok refactored things now let's see how this baby flies on the circle-ci side...

Remi-Gau commented:
Note that given that one of the preivous build did not complete because we hit the circle CI "wall time" I think we may have to move to github actions that will give us 6 hours before we get cancelled. I am not as good with caching things in GHA action though...

https://app.circleci.com/pipelines/github/bids-apps/freesurfer/82/workflows/bc88a5ed-4c03-4c1e-ba7a-dd0b33577e60/jobs/340

But let's wait and see how things are going on this build... Not holding my breath.

Remi-Gau commented:
note that the image is using an antiquated bids-validator version (0.19.8 when the current one is 1.12.0): may be worth bumping this up.

Remi-Gau commented:
ok the image build is failing because of some conda hanging: https://app.circleci.com/pipelines/github/bids-apps/freesurfer/87/workflows/d6fe415d-5abd-427d-95c9-792ab1bf0073/jobs/355

trying locally

Remi-Gau commented:
The only that changes in the bash script used to generate the different recipe is the freesurfer version, no?

I think this should be refactored too

Remi-Gau commented:
@Shotgunosine can you check if you can build the image using the docker recipe: it seems to fail in CI and for me locally.

May need to adapt the neurodocker command

Shotgunosine committed:
refactor bash script

Shotgunosine commented:
If hangs forever solving the environment for me locally as well. I'm trying to see if there's a way to install pandas that doesn't hang, but I haven't found anything yet. Might need to add mamba to neurodocker.

Remi-Gau commented:
> If hangs forever solving the environment for me locally as well. I'm trying to see if there's a way to install pandas that doesn't hang, but I haven't found anything yet. Might need to add mamba to neurodocker.

maybe specifying the exact version of each package may help?

Shotgunosine committed:
use mamba, bumb bids-validator version

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Shotgunosine commented:
I was able to get it to build with a local version of neurodocker that I added a mamba option to, and I've submitted a pull request for that: https://github.com/ReproNim/neurodocker/pull/562

Note that for generate_freesurfer_images.sh I actually used my local neurodocker image, not that sha that's in there now.

Shotgunosine committed:
update neurodocker

Shotgunosine committed:
parameterize license directory to avoide cache collision

Shotgunosine committed:
change entrypoint to python

Shotgunosine committed:
update versions in readme

Shotgunosine commented:
@Remi-Gau I've got no idea why circle is trying to do anything with /home/circleci/license/license.txt (which is the source of the failure here: /home/circleci/license/license.txt). As far as I can tell I've changed all references to that directory and the build ran past it when I changed the entry point to python. Do you have any ideas? I'm stumped.

Shotgunosine committed:
deal with license

Remi-Gau commented:
> Edit: Nevermind , figured it out.

🚀

Remi-Gau commented on .circleci/config.yml:
```diff
@@ -60,51 +70,65 @@ jobs:
+            -v ${HOME}/outputs1:/outputs \
+            --tmpfs /tmp --tmpfs /var/tmp \
+              bids/${CIRCLE_PROJECT_REPONAME}_fs<< parameters.version >> \
+                /bids_dataset
```
```suggestion
                /bids_dataset \
```

Remi-Gau committed:
Update .circleci/config.yml

Remi-Gau commented:
will make a dummy change to retrigger circle ci

Remi-Gau commented:
getting somse validator error: will bump its version

Remi-Gau commented:
> getting somse validator error: will bump its version

totally missed that you had done that and completely refactored the bash script: you've been busy

Shotgunosine committed:
bump node to 18.x

Shotgunosine committed:
trigger circle?

Shotgunosine commented:
Bah, looks like circleci webhooks are broken at the moment. Is there anyone who can trigger a manual build?

Remi-Gau commented:
> Is there anyone who can trigger a manual build?

just did: let's see how this one goes

Remi-Gau commented:
see if we can create specific dependencies between the steps of the circle ci so that test for fs6 starts when the build for fs6 is done.

Currently the tests can only start when both builds are done.

![image](https://github.com/bids-apps/freesurfer/assets/6961185/29581cf4-2b35-4a48-94d9-0486a7733745)

Shotgunosine committed:
add python2 for fs6 and fix installation directory for fs7

Shotgunosine committed:
fix fs7 install dir

Shotgunosine committed:
parameterize job names

Shotgunosine committed:
paramterize job dependencies

Shotgunosine commented:
> see if we can create specific dependencies between the steps of the circle ci so that test for fs6 starts when the build for fs6 is done.
> 
> Currently the tests can only start when both builds are done.
> 
> ![image](https://user-images.githubusercontent.com/6961185/264158564-29581cf4-2b35-4a48-94d9-0486a7733745.png)

Yep, found the docs on matrix and they explain how to set it up. The circleci graph makes it looks like the dependencies are correct now.

Remi-Gau commented on Dockerfile:
shouldn't we just delete the Dockerfile?

Remi-Gau commented:
> Yep, found the [docs on matrix](https://circleci.com/docs/configuration-reference/#matrix) and they explain how to set it up. The circleci graph makes it looks like the dependencies are correct now.

nice!!!

thanks for looking into this

Shotgunosine committed:
pin pandas as 1.5.3

Shotgunosine commented:
@Remi-Gau, the tests are failing because they can't find anything at [/home/circleci/data/ds114_test1_freesurfer_precomp_v6.0.0/00_group*](https://app.circleci.com/pipelines/github/bids-apps/freesurfer/103/workflows/f4ee1e9a-63ab-4913-b0a6-149f25573870/jobs/419?invite=true#step-109-3496_112). That part of the circleconfig predates this PR. Do you know what's going on with that?

Remi-Gau committed:
rm Dockerfile

Remi-Gau commented:
get the same bug locally:

```bash
# get download script
wget https://raw.githubusercontent.com/bids-apps/maintenance-tools/main/utils/get_data_from_osf.sh

# get gata into home dir from https://osf.io/9q7dv/
bash get_data_from_osf.sh ds114_test1_freesurfer_precomp_v6.0.0 

# create target dir
mkdir -p ${HOME}/outputs1 ${HOME}/outputs2

# try to copy
sudo mv ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0.0/00_group* ${HOME}/outputs1/
```
no group folder

```bash
$ ls ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0.0/       
  
fsaverage  lh.EC_average  rh.EC_average  sub-01  sub-02
```

let me dig into the history of the repo to see what was happening in the good old days

Remi-Gau commented:
here is the circle ci "step" that was used before I started working in the circle-ci:

https://github.com/bids-apps/freesurfer/blob/8c364a62ffbdef3b3e4f81a1c6bf4de5e9c2b5cd/.circleci/config.yml#L66C67-L66C67

more readable

```bash
docker run -ti --rm --read-only \
  -v ~/license/license.txt:/license.txt \
  --tmpfs /tmp --tmpfs /var/tmp \
  -v ${HOME}/data/ds114_test1:/bids_dataset \
  -v ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0.0:/outputs \
    bids/${CIRCLE_PROJECT_REPONAME} /bids_dataset /outputs group2 \
      --license_file=/license.txt && \
mkdir -p ${HOME}/outputs1/ && \
sudo mv ${HOME}/data/ds114_test1_freesurfer_precomp_v6.0.0/00_group* ${HOME}/outputs1/ && \
cat ${HOME}/outputs1/00_group2_stats_tables/lh.aparc.thickness.tsv && \
cat ${HOME}/outputs1/00_group2_stats_tables/euler.tsv
```

Remi-Gau committed:
it is always a path issue

Remi-Gau commented:
I think I it is "just" a path issue: trying to fix in https://github.com/bids-apps/freesurfer/pull/67/commits/765a747844da7745a49b19c1087a8098ce213589

Shotgunosine commented:
@Remi-Gau, it passed! Is there anything else we should fix for this PR?

Remi-Gau commented:
woohoo!!!

I will merge and make a new release: this should trigger a push to docherhub.

bpinsard commented:
Wow, glad my ping triggered a new release! Thanks for all the hard work! Waiting on the dockerhub push and will make a PR to repronim, can't wait to try it out!

Remi-Gau commented:
> Wow, glad my ping triggered a new release! Thanks for all the hard work! Waiting on the dockerhub push and will make a PR to repronim, can't wait to try it out!

FYI it is already on dockerhub but it is a bit of a mess at the moment:

see here
https://github.com/bids-apps/freesurfer/issues/75#issuecomment-1699911795

any thoughts on what would make most sense for users?
